lib/logstash/inputs/gelf.rb

```diff
@@ -33,6 +33,13 @@ class LogStash::Inputs::Gelf < LogStash::Inputs::Base
   # ports) may require root to use.
   config :port, :validate => :number, :default => 12201
 
+  # The GELF protocol (TCP or UDP).
+  config :protocol, :validate => :string, :default => "UDP"
```

Comment:
You can use `:validate => [ "UDP", "TCP" ]` to enforce a specific set of available values.